src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderEmailService.java

```diff
@@ -863,15 +701,60 @@ private void sendEmailToSolicitorOrNotifyParties(List<DynamicMultiselectListElem
                     );
                 } else {
                     log.info("*** Send email/post notifications to parties ***");
-                    sendNotificationsToParty(
-                        caseData,
-                        partyDataOptional.get(),
-                        authorisation,
-                        dynamicDataForEmail,
-                        orderDocuments,
-                        bulkPrintOrderDetails,
-                        SendgridEmailTemplateNames.SERVE_ORDER_APPLICANT_RESPONDENT
-                    );
+                    if (ContactPreferences.email.equals(partyData.getContactPreferences())
+                        && isPartyProvidedWithEmail(partyData) && CaseUtils.isAccessEnabled(partyDataOptional.get())) {
```

Comment:
Are we not having existing method to determine new/final/new final order flags? if yes can we use it else move this to new method